### DISCUSSION TITLE
Need sample code or steps to configure CAN receive DMA on lpc55s16 microcontroller

### DISCUSSION THREAD

**Discussion by aravinth-kalai** [zephyrproject-rtos/zephyr#71478] [Q&A] [answered]
Hello everyone,

Currently I'm working on CAN receive with DMA, I have searched on zephyr repository I didn't find any sample code to configure DMA, if you have any reference link to configure the DMA for this microcontroller or some other microcontroller please share, it will be really helpful.

Thanks in advance !

**Comment by DerekSnell** [comment]
Hi @aravinth-kalai ,

I have also not found a Zephyr example using DMA with CAN, nor am I aware of one.  And it does not appear NXP's MCUXpresso SDK includes a CAN example with DMA.

Hi @henrikbrixandersen , you work with CAN more than I do.  Are you aware of a Zephyr example using DMA?

Best regards

**Comment by henrikbrixandersen** [comment, marked as answer]
The Bosch M_CAN IP core as used in the LPC55xxx series only supports DMA via an extension-module called DMU (DMA Interface Unit). As far as I know, the LPC55xxx series does not include the DMU.

I am looking into providing DMA support for the CAN drivers where it can be done (e.g. FlexCAN FD with Enhanced FIFO support), but that's another story.

@aravinth-kalai I am curious to hear how you reached to conclusion, you'd need use DMA along with the LPC55xxx MCAN?
